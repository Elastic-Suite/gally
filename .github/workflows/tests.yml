name: Tests

on:
  workflow_call:
    inputs:
      version:
        description: "Version of Gally's packages (ex: master or 1.0.x)"
        type: string
        default: 'master'
        required: false
      last_published_version:
        description: "Last release of the current version"
        required: false
        default: "dev-master"
        type: string
    secrets:
      COMPOSER_AUTH:
        required: true

jobs:
  tests:
    if: github.event.pull_request.draft == false
    name: Tests
    runs-on: ubuntu-latest
    
    env:
      back_branch: ${{ inputs.version }}
      front_branch: ${{ inputs.version == 'master' && 'main' || inputs.version }}
      composer_version: ${{ inputs.version == 'master' && 'dev-master' || (contains(inputs.version, 'x') && format('{0}-dev', inputs.version) || inputs.version) }}
    
    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          ref: ${{ env.back_branch }}
      - name: Checkout Gally admin
        uses: actions/checkout@v3
        with:
          repository: 'Elastic-Suite/gally-admin'
          ref: ${{env.front_branch}}
          path: front/gally-admin
      - name: update gally standard package
        working-directory: api
        run: composer require "gally/gally-standard:${{ env.composer_version }} as ${{ inputs.last_published_version }}"
      - name: Add premium package repository
        uses: php-actions/composer@v6
        with:
          php_version: "8.1"
          args: --working-dir api
          command: config repositories.gally-premium composer https://packagist.smile.fr/gally
        env:
          COMPOSER_AUTH: ${{ secrets.COMPOSER_AUTH }}
      - name: Install premium package
        uses: php-actions/composer@v6
        with:
          php_version: "8.1"
          args: --no-cache --working-dir api
          command: require gally/gally-premium:${{ env.composer_version }}
        env:
          COMPOSER_AUTH: ${{ secrets.COMPOSER_AUTH }}
      - name: Pull images
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml pull --ignore-pull-failures || true
      - name: Start services
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml up --build -d --wait
      - name: Check HTTP reachability
        id: http_reachability
        run: curl -v -o /dev/null http://localhost
      - name: HTTP Failure
        if: ${{ failure() && steps.tests.http_reachability == 'failure' }}
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml logs
      - name: Check HTTPS reachability
        id: https_reachability
        run: curl  -vk -o /dev/null https://localhost
      - name: HTTPS Failure
        if: ${{ failure() && steps.tests.https_reachability == 'failure' }}
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml logs
      - name: Create test database
        run: |
          docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=off -T php bin/console -e test doctrine:database:create
          docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=off -T php bin/console -e test doctrine:migrations:migrate --no-interaction
      - name: Doctrine Schema Validator
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=off -T php bin/console doctrine:schema:validate --skip-sync
      - name: Fixtures load
        run: |
          docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=off -T php bin/console gally:index:clear --no-interaction
          docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=off -T php bin/console hautelook:fixtures:load --no-interaction -vv
          docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=off -T php bin/console doctrine:fixtures:load --append --no-interaction
      - name: Coverage
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -e XDEBUG_MODE=coverage -T php bin/phpunit --coverage-clover=coverage/coverage.xml vendor/gally

      - name: Cleanup coverage file
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -T php sed -i 's:<file name="/srv/:<file name="/:' coverage/coverage.xml

      - if: ${{ github.event_name == 'pull_request' }}
        name: Download artifact
        uses: dawidd6/action-download-artifact@v2.14.1
        continue-on-error: true
        id: base_coverage_report
        with:
          workflow: .github/workflows/ci.yml
          branch: ${{ github.base_ref }}
          name: coverage-report
          path: coverage/base # Retrieve previous coverage file at the root directory, not inside api/coverage

      - name: Backend Coverage Report as Comment (Compared to a base)
        if: ${{ github.event_name == 'pull_request' && steps.base_coverage_report.outcome == 'success' }}
        uses: lucassabreu/comment-coverage-clover@main
        continue-on-error: true
        with:
          with-chart: false
          file: api/coverage/coverage.xml
          base-file: coverage/base/coverage.xml
          max-line-coverage-decrease: 10.0
          max-method-coverage-decrease: 10.0
          min-line-coverage: 75
          min-method-coverage: 75
          signature: "Backend Coverage compared to target branch"

      - name: Backend Coverage Report as Comment (Clean)
        uses: lucassabreu/comment-coverage-clover@v0.3.0
        if: ${{ (github.event_name == 'pull_request' && steps.base_coverage_report.outcome != 'success') || github.event_name == 'push'}}
        with:
          with-chart: false
          file: api/coverage/coverage.xml
          signature: "Backend Coverage of this branch"

      - uses: actions/upload-artifact@v2
        with:
          name: coverage-report
          path: api/coverage/coverage.xml

      - name: Jest
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml exec -T pwa yarn test:ci

      - name: Frontend Coverage Report
        uses: 5monkeys/cobertura-action@v12
        if: ${{ github.event_name == 'pull_request' }}
        with:
          repo_token: "${{ secrets.GITHUB_TOKEN }}"
          report_name: "Frontend Coverage of this branch"
          path: front/example-app/coverage/cobertura-coverage.xml
          minimum_coverage: 1
          fail_below_threshold: false

      - name: "[End] Job failed, gathering logs"
        if: ${{ failure() }}
        run: docker compose -f docker-compose.yml -f docker-compose.ci.yml logs
