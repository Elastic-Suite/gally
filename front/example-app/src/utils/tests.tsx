import { RenderOptions, render, renderHook } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ReactNode } from 'react'
import { api } from 'gally-admin-shared'

import TestProvider from './TestProvider'

export type IExtendedRenderOptions = Omit<RenderOptions, 'queries'>

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <MemoryRouter>
        <TestProvider api={api}>{children}</TestProvider>
      </MemoryRouter>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <MemoryRouter>
        <TestProvider api={api}>{children}</TestProvider>
      </MemoryRouter>
    )
  }
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
