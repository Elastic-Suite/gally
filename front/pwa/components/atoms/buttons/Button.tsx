import { MouseEvent, useRef, useState } from 'react'
import { styled } from '@mui/system'
import {
  ButtonProps,
  CircularProgress,
  MenuItem,
  Menu as MuiMenu,
} from '@mui/material'
import { IOptions } from 'shared'

import { PrimaryButton, SecondaryButton, TertiaryButton } from './Button.styled'

declare module '@mui/material' {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  interface MenuProps {
    minWidth?: number
  }
}

const menuProps = ['minWidth']
const Menu = styled(MuiMenu, {
  shouldForwardProp: (prop: string) => !menuProps.includes(prop),
})(({ minWidth, theme }) => ({
  '& .MuiPaper-root': {
    margin: '4px 0',
    border: `1px solid ${theme.palette.colors.neutral['300']}`,
    borderRadius: 8,
    overflow: 'auto',
    boxSizing: 'border-box',
    outline: 0,
    boxShadow: 'none',
    minWidth,
  },
  '& .MuiList-root': {
    padding: 0,
  },
  '& .MuiButtonBase-root': {
    fontFamily: 'Inter',
    padding: '8px 12px',
    fontWeight: 400,
    fontSize: 12,
    lineHeight: '18px',
    color: theme.palette.colors.neutral['800'],
  },
  '& .MuiButtonBase-root:hover': {
    backgroundColor: theme.palette.colors.neutral['200'],
  },
}))

interface IProps extends ButtonProps {
  display: 'primary' | 'secondary' | 'tertiary'
  loading?: boolean
  onSelect?: (value: unknown) => void
  options?: IOptions<unknown>
}

function Button(props: IProps): JSX.Element {
  const {
    children,
    disabled,
    display,
    id,
    loading,
    onClick,
    onSelect,
    options,
    style,
    ...buttonProps
  } = props
  const buttonRef = useRef<HTMLButtonElement>(null)
  const [anchorEl, setAnchorEl] = useState(null)
  const open = Boolean(anchorEl)

  let Component = PrimaryButton
  if (display === 'secondary') {
    Component = SecondaryButton
  } else if (display === 'tertiary') {
    Component = TertiaryButton
  }

  let styles = style
  if (loading && buttonRef.current) {
    const { height, width } = buttonRef.current.getBoundingClientRect()
    styles = {
      ...style,
      ...(loading && {
        width,
        height,
      }),
    }
  }

  function handleOpen(event: MouseEvent<HTMLButtonElement>): void {
    setAnchorEl(event.currentTarget)
    if (onClick) {
      onClick(event)
    }
  }

  function handleClose(): void {
    setAnchorEl(null)
  }

  function handleClickOption(value: unknown) {
    return () => {
      onSelect(value)
      setAnchorEl(null)
    }
  }

  return (
    <>
      <Component
        {...buttonProps}
        disabled={disabled || loading}
        onClick={options ? handleOpen : onClick}
        ref={buttonRef}
        style={styles}
      >
        {loading ? <CircularProgress color="inherit" size="20px" /> : children}
      </Component>
      {Boolean(options) && (
        <Menu
          anchorEl={anchorEl}
          componentsProps={{
            backdrop: { style: { backgroundColor: 'transparent' } },
          }}
          id={id}
          open={open}
          onClose={handleClose}
          MenuListProps={{
            'aria-labelledby': id,
          }}
          minWidth={anchorEl ? anchorEl.getBoundingClientRect().width : 0}
        >
          {options.map((option) => (
            <MenuItem
              key={option.id ?? String(option.value)}
              onClick={handleClickOption(option.value)}
            >
              {option.label}
            </MenuItem>
          ))}
        </Menu>
      )}
    </>
  )
}

Button.defaultProps = {
  display: 'primary',
  variant: 'contained',
}

export default Button
