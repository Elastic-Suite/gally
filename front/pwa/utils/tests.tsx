import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { ReactNode } from 'react'
import { Bundle, IUser, api } from 'shared'

import { AppStore, RootState, setupStore } from '~/store'

import AppProvider from '~/components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const timestamp = Math.floor(Date.now() / 1000)
const expires = 8 * 60 * 60 // 8 hours

const user = {
  iat: timestamp,
  exp: timestamp + expires,
  roles: ['ROLE_ADMIN', 'ROLE_CONTRIBUTOR'],
  username: 'admin@example.com',
} as IUser

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: ['https://localhost/products'],
  },
  user: {
    requestedPath: '',
    token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2NTk5NTM1NDQsImV4cCI6MTY1OTk4MjM0NCwicm9sZXMiOlsiUk9MRV9BRE1JTiIsIlJPTEVfQ09OVFJJQlVUT1IiXSwidXNlcm5hbWUiOiJhZG1pbkBleGFtcGxlLmNvbSJ9.ZNE2SHmtm9WgWEMmKGMjpcfEym2ysxC_CVcLtzlpuT3jlT0Zyz4VjoQHdTav9I7s9b2QzpeDNt3Sfrbka8IHcVb_TvZle_vII-TkQGal37RtXxKk_EMMQQlzLRO3zDWYy2I0ZPXv6iMlaSWOya4egOG2s1_tzFf4MnNNm2zEQbjzAtZ_Ij9UBMlSxK19AHvHlKU9AelA-NepjUdl6f4GxYPnyIwZw3PS294Wtffj7Kzw1ZZywDljl1xN_flg70cuO0GGkUqBNHrf7zBjPmXbAZZDAefoyQfVqB2v4GzDJfO847XymvBlHtCCjWx0ThK5afyOGL2MAAZkl-AoNjMcjA',
    user,
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
