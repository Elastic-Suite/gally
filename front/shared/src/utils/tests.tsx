import { RenderOptions, render, renderHook } from '@testing-library/react'
import { ReactNode } from 'react'

import { schemaContext } from '../contexts'
import { api } from '../mocks'

export type IExtendedRenderOptions = Omit<RenderOptions, 'queries'>

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <schemaContext.Provider value={api}>{children}</schemaContext.Provider>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <schemaContext.Provider value={api}>{children}</schemaContext.Provider>
    )
  }
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
