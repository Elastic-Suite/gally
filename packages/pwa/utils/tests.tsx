import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { ReactChild } from 'react'

import AppProvider from '~/components/stateful-providers/AppProvider/AppProvider'
import { api } from 'shared'
import { AppStore, RootState, setupStore } from '~/store'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactChild }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider api={api}>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactChild }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider api={api}>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
