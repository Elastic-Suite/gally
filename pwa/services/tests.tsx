import { RenderOptions, render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { PreloadedState } from '@reduxjs/toolkit'
import { ReactChild } from 'react'

import AppProvider from '~/components/AppProvider'
import { AppStore, RootState, setupStore } from '~/store'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactChild }): JSX.Element {
    return <AppProvider store={store}>{children}</AppProvider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactChild }): JSX.Element {
    return <AppProvider store={store}>{children}</AppProvider>
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
