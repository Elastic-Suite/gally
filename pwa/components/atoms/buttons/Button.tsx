import { MouseEvent, useState } from 'react'
import { styled } from '@mui/system'
import { ButtonProps, MenuItem, Menu as MuiMenu } from '@mui/material'
import { IOptions } from '~/types'

import { PrimaryButton, SecondaryButton, TertiaryButton } from './Button.styled'

declare module '@mui/material' {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  interface MenuProps {
    minWidth?: number
  }
}

const menuProps = ['minWidth']
const Menu = styled(MuiMenu, {
  shouldForwardProp: (prop: string) => !menuProps.includes(prop),
})(({ minWidth, theme }) => ({
  '& .MuiPaper-root': {
    margin: '4px 0',
    border: `1px solid ${theme.palette.colors.neutral['300']}`,
    borderRadius: 8,
    overflow: 'auto',
    boxSizing: 'border-box',
    outline: 0,
    boxShadow: 'none',
    minWidth,
  },
  '& .MuiList-root': {
    padding: 0,
  },
  '& .MuiButtonBase-root': {
    fontFamily: 'Inter',
    padding: '8px 12px',
    fontWeight: 400,
    fontSize: 12,
    lineHeight: '18px',
    color: theme.palette.colors.neutral['800'],
  },
  '& .MuiButtonBase-root:hover': {
    backgroundColor: theme.palette.colors.neutral['200'],
  },
}))

interface IProps extends ButtonProps {
  display: 'primary' | 'secondary' | 'tertiary'
  onSelect?: (value: unknown) => void
  options?: IOptions<unknown>
}

function Button(props: IProps): JSX.Element {
  const { display, id, onClick, onSelect, options, ...buttonProps } = props
  const [anchorEl, setAnchorEl] = useState(null)
  const open = Boolean(anchorEl)

  let Component = PrimaryButton
  if (display === 'secondary') {
    Component = SecondaryButton
  } else if (display === 'tertiary') {
    Component = TertiaryButton
  }

  function handleOpen(event: MouseEvent<HTMLButtonElement>): void {
    setAnchorEl(event.currentTarget)
    if (onClick) {
      onClick(event)
    }
  }

  function handleClose(): void {
    setAnchorEl(null)
  }

  function handleClickOption(value: unknown) {
    return () => {
      onSelect(value)
      setAnchorEl(null)
    }
  }

  return (
    <>
      <Component {...buttonProps} onClick={options ? handleOpen : onClick} />
      {Boolean(options) && (
        <Menu
          anchorEl={anchorEl}
          componentsProps={{
            backdrop: { style: { backgroundColor: 'transparent' } },
          }}
          id={id}
          open={open}
          onClose={handleClose}
          MenuListProps={{
            'aria-labelledby': id,
          }}
          minWidth={anchorEl ? anchorEl.getBoundingClientRect().width : 0}
        >
          {options.map((option) => (
            <MenuItem
              key={option.id ?? String(option.value)}
              onClick={handleClickOption(option.value)}
            >
              {option.label}
            </MenuItem>
          ))}
        </Menu>
      )}
    </>
  )
}

Button.defaultProps = {
  display: 'primary',
  variant: 'contained',
}

export default Button
