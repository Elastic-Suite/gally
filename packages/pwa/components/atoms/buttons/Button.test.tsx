import { renderWithProviders } from '~/utils/tests'

import Button from './Button'

describe('Button match snapshot', () => {
  it('PrimaryButton', () => {
    const { container } = renderWithProviders(
      <Button disabled={false} endIcon="" size="medium" startIcon="">
        Hello world
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  it('SecondaryButton', () => {
    const { container } = renderWithProviders(
      <Button
        display="secondary"
        disabled={false}
        endIcon=""
        size="medium"
        startIcon=""
      >
        Hello world
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  it('TertiaryButton', () => {
    const { container } = renderWithProviders(
      <Button
        display="tertiary"
        disabled={false}
        endIcon=""
        size="medium"
        startIcon=""
      >
        Hello world
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  it('PrimaryButtonDisabledTrue', () => {
    const { container } = renderWithProviders(
      <Button disabled endIcon="" size="medium" startIcon="">
        Hello world
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  it('SecondaryButtonDisabledTrue', () => {
    const { container } = renderWithProviders(
      <Button
        display="secondary"
        disabled
        endIcon=""
        size="medium"
        startIcon=""
      >
        Hello world
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  it('TertiaryButtonDisabledTrue', () => {
    const { container } = renderWithProviders(
      <Button display="tertiary" disabled endIcon="" size="medium" startIcon="">
        Hello world
      </Button>
    )
    expect(container).toMatchSnapshot()
  })
})
