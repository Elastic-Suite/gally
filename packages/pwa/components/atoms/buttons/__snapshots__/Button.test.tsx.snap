// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot PrimaryButton 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13u4ebv-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Hello world
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button match snapshot PrimaryButtonDisabledTrue 1`] = `
<div>
  <button
    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13u4ebv-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    Hello world
  </button>
</div>
`;

exports[`Button match snapshot SecondaryButton 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wby1ea-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Hello world
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button match snapshot SecondaryButtonDisabledTrue 1`] = `
<div>
  <button
    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wby1ea-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    Hello world
  </button>
</div>
`;

exports[`Button match snapshot TertiaryButton 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13yl9aj-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Hello world
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button match snapshot TertiaryButtonDisabledTrue 1`] = `
<div>
  <button
    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13yl9aj-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    Hello world
  </button>
</div>
`;
